Source/Core/Common/ArmCPUDetect.cpp

```diff
@@ -232,15 +250,14 @@ void CPUInfo::Detect()
   num_cores = std::max(static_cast<int>(std::thread::hardware_concurrency()), 1);
 
 #ifdef __APPLE__
-  SysctlByName(&model_name, "machdep.cpu.brand_string");
-
-  // M-series CPUs have all of these
-  // Apparently the world has accepted that these can be assumed supported "for all time".
-  // see https://github.com/golang/go/issues/42747
-  bAES = true;
-  bSHA1 = true;
-  bSHA2 = true;
-  bCRC32 = true;
+  model_name = ReadSysctlByNameString("machdep.cpu.brand_string").value_or("(not found)");
+
+  // Specific strings to use as as per Apple:
+  // https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics
+  bAES = IsSysctlByNameFeaturePresent("hw.optional.arm.FEAT_AES");
```

Comment:
Sadly - out of these four optional sysctls, macOS 11.0 only has `hw.optional.armv8_crc32`. (This sysctl seems to have been present since iOS 12, likely because A7/A8/A9 lack CRC32 being ARMv8.0.)

[Apple also seems to currently assume](https://github.com/apple-oss-distributions/xnu/blob/e6231be02a03711ca404e5121a151b24afbff733/osfmk/arm/commpage/commpage.c#L409-L421) that all of their ARMv8 cores have SHA-1/SHA-256/AES - would it be worth defaulting the feature to true if the sysctl isn't present?